custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,461 @@
+    "off": (),
+    "heat": ("setpoint_heating",),
+    "cool": ("setpoint_cooling",),
+    "auto": ("setpoint_heating", "setpoint_cooling"),
+    "aux heat": ("setpoint_heating",),
+    "furnace": ("setpoint_furnace",),
+    "dry air": ("setpoint_dry_air",),
+    "moist air": ("setpoint_moist_air",),
+    "auto changeover": ("setpoint_auto_changeover",),
+    "heat econ": ("setpoint_eco_heating",),
+    "cool econ": ("setpoint_eco_cooling",),
+    "away": ("setpoint_away_heating", "setpoint_away_cooling"),
+    "full power": ("setpoint_full_power",),
+}
+
+# strings, OZW and/or qt-ozw does not send numeric values
+HVAC_CURRENT_MAPPINGS = {
+    "idle": CURRENT_HVAC_IDLE,
+    "heat": CURRENT_HVAC_HEAT,
+    "pending heat": CURRENT_HVAC_IDLE,
+    "heating": CURRENT_HVAC_HEAT,
+    "cool": CURRENT_HVAC_COOL,
+    "pending cool": CURRENT_HVAC_IDLE,
+    "cooling": CURRENT_HVAC_COOL,
+    "fan only": CURRENT_HVAC_FAN,
+    "vent / economiser": CURRENT_HVAC_FAN,
+    "off": CURRENT_HVAC_OFF,
+}
+
+DEFAULT_HVAC_MODES = [
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_DRY,
+    HVAC_MODE_OFF,
+    HVAC_MODE_AUTO,
+]
+
+# ZW 1.6 mode to HASS HVAC_MODE Mapping
+HVAC_MODE_MAPPINGS = {
+    0x00: HVAC_MODE_OFF,
+    0x01: HVAC_MODE_HEAT,
+    0x02: HVAC_MODE_COOL,
+    0x03: HVAC_MODE_HEAT_COOL,
+    0x04: HVAC_MODE_HEAT,
+    0x06: HVAC_MODE_FAN_ONLY,
+    0x07: HVAC_MODE_HEAT,
+    0x08: HVAC_MODE_DRY,
+    0x0A: HVAC_MODE_HEAT_COOL,
+    0x0B: HVAC_MODE_HEAT,
+    0x0C: HVAC_MODE_COOL,
+    0x0D: HVAC_MODE_HEAT_COOL,
+    0x0F: HVAC_MODE_HEAT_COOL,
+}
+
+ZW_HVAC_MODE_HEAT = 0x01
+ZW_HVAC_MODE_COOL = 0x02
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Z-Wave Climate from Config Entry."""
+
+    @callback
+    def async_add_climate(values):
+        """Add Z-Wave Climate."""
+        climate = None
+        if values.primary.command_class == CommandClass.THERMOSTAT_SETPOINT:
+            climate = ZWaveClimateSingleSetpoint(values)
+        elif values.primary.command_class == CommandClass.THERMOSTAT_MODE:
+            climate = ZWaveClimateMultipleSetpoint(values)
+
+        if climate is not None:
+            async_add_entities([climate])
+
+    hass.data[DOMAIN][config_entry.entry_id][DATA_UNSUBSCRIBE].append(
+        async_dispatcher_connect(hass, "zwave_new_climate", async_add_climate)
+    )
+    await hass.data[DOMAIN][config_entry.entry_id]["mark_platform_loaded"]("climate")
+
+
+class ZWaveClimateBase(ZWaveDeviceEntity, ClimateDevice):
+    """Representation of a Z-Wave Climate device."""
+
+    def __init__(self, values):
+        """Initialize the Z-Wave climate device."""
+        super().__init__(values)
+        self._target_temperature = None
+        self._target_temperature_range = (None, None)
+        self._current_temperature = None
+        self._hvac_action = None
+        self._hvac_list = None
+        self._hvac_mode = None
```

Comment:
We should rename `self._hvac_mode` to `self._zw_hvac_mode` to make it clear that it's a zwave value and not a home assistant value.